contrib/systems.md/thermostat.md

```diff
@@ -46,23 +81,135 @@ mod THERMOSTAT is
                    => < TIME , TMP , off >
                    if TMP < min + bound = false .
 
-    rl [delaying] : { TIME + 1 , TMP , DM }
-                 => < TIME     , TMP , DM > .
+   crl [delaying] : { TIME     , TMP , DM }
+                 => < TIME - 1 , TMP , DM >
+                 if 0 < TIME = true .
 
     rl [delay-over] : { 0 , TMP , delay(IM) }
                    => < 0 , TMP , IM        > .
 ```
 
+The thermostat `heat-rate` is determined as a function of the thermostat parameters.
+
+```maude
+    op heat-rate : Mode TMP$Ring -> TMP$Ring .
+    ------------------------------------------
+    eq heat-rate(MODE, TMP) = (TMP + source(MODE)) - drain(TMP) .
+```
+
 The following are parameters which must be filled in for your particular thermostat.
 
 ```maude
-   ops min max bound : -> Nat .
-   ----------------------------
+    ops min max bound : -> TMP$Ring .
+    ---------------------------------
 
-   ops time-until : InMode -> Nat .
-   --------------------------------
+    op time-until : InMode -> TIME$Ring .
+    -------------------------------------
 
-    op heat-rate : Mode Nat -> Nat .
-    --------------------------------
+    op source : Mode -> TMP$Ring .
+    ------------------------------
+
+    op drain : TMP$Ring -> TMP$Ring .
+    ---------------------------------
+endm
+```
+
+Thermostat Instantiation
+------------------------
+
+### Instantiation to FVP-INT
+
+ ```maude
+load ../tools/fvp/numbers.maude
+
+view RingInt from UNITAL-ORDERED-RING to FVP-INT-PRED is
+sort RingBool to Bool .
+    sort Ring     to Int  .
+endv
+
+mod THERMOSTAT-INT is protecting THERMOSTAT{RingInt, RingInt} . endm
+```
+
+The following is an example thermostat over the integers.
+
+```maude
+mod THERMOSTAT-INT-COMFORTABLE is
+    extending THERMOSTAT-INT + FVP-NUMBERS .
+    var TMP : Int .
+```
+
+Here we setup a thermostat which tries to stay between `18` and `26`, and will switch states when `3` away from a boundary temperature.
+
+```maude
+    eq min   = 10 + 8      [variant] .
+    eq max   = 10 + 10 + 6 [variant] .
+    eq bound = 3           [variant] .
+```
+
+The thermostat takes `4` seconds to turn on, and `2` seconds to turn off.
+
+```maude
+    eq time-until(on)  = 4 [variant] .
+    eq time-until(off) = 2 [variant] .
+```
+
+When turning on/off, the heater is effective for `1` unit temperature per second.
+When on, the heater produces `5` units temperature per second, and when off produces nothing.
+Every second, `3` units of temperature are drained to the environment.
+
+```maude
+    eq source(on)         = 5 [variant] .
+    eq source(off)        = 0 [variant] .
+    eq source(delay(on))  = 2 [variant] .
+    eq source(delay(off)) = 2 [variant] .
+
```

Comment:
The comments seem be to in disagreement with the code, during the delay period, the comments say `1` unit of heat is produced, but in the code, it has `2`.

Reply:
Fixed, also added units.